subsys/bluetooth/host/conn.c

```diff
@@ -2249,7 +2249,7 @@ int bt_conn_auth_pairing_confirm(struct bt_conn *conn)
 }
 #endif /* CONFIG_BT_SMP || CONFIG_BT_BREDR */
 
-u8_t bt_conn_get_id(struct bt_conn *conn)
+u8_t bt_conn_get_index(struct bt_conn *conn)
 {
 	return conn - conns;
```

Comment:
There are no checks at all here for invalid or NULL pointer, which will trigger a fault. Maybe we could add at the very minimum an `__ASSERT()` ?

Reply:
At most an __ASSERT I'd say, since otherwise we're complicating all callers (it's no-longer appropriate to pass the return value as an array index without first checking the return value)

Reply:
Anyway, it's not directly related to the patch, so should be a separate commit IMO. It *is* a valid thing to raise though since now there can be external callers who's code validity we don't have the same control of as we do for internal code.